src/hooks/search/location.ts

```diff
@@ -23,12 +23,29 @@ const useLocationSearch = (callback: () => void) => {
         useState<LocationSelection>();
     const [geolocationValidating, setGeolocationValidating] = useState(false);
 
-    const { data: locationResults, isValidating: locationValidating } = useSWR(
+    const {
+        data: rawLocationResults = [],
+        isValidating: locationValidating,
+    }: {
+        data?: google.maps.places.AutocompletePrediction[];
+        isValidating: boolean;
+    } = useSWR(
         locationQuery ? `search=${locationQuery}` : null,
         locationFetcher,
         swrOptions
     );
 
+    // Remove large result types since we're comparing coordinates
+    const locationResults = useMemo(
```

Comment:
saw you have some other TODO/refactors here, so we can probably address this when we have more time, but you might be able to adjust the `types` param in `/api/location` query to only include certain type of result sets to avoid this filtering

https://developers.google.com/maps/documentation/places/web-service/supported_types